scripts/exrrfs_upp.sh

```diff
@@ -1,25 +1,78 @@
 #!/usr/bin/env bash
 declare -rx PS4='+ $(basename ${BASH_SOURCE[0]:-${FUNCNAME[0]:-"Unknown"}})[${LINENO}]${id}: '
 set -x
-cpreq=${cpreq:-cpreq}
 
-cd ${DATA}/${FHR}
-fhr=${FHR:1:2} # remove leading zeros
-CDATEp=$($NDATE ${fhr} ${CDATE} )
-timestr=$(date -d "${CDATEp:0:8} ${CDATEp:8:2}" +%Y-%m-%d_%H.%M.%S)
-timestr2=$(date -d "${CDATEp:0:8} ${CDATEp:8:2}" +%Y-%m-%d_%H:%M:%S)
+ulimit -s unlimited
+ulimit -v unlimited
+ulimit -a
 
-if [[ -z "${ENS_INDEX}" ]]; then
-  ensindexstr=""
-else
-  ensindexstr="/mem${ENS_INDEX}"
-fi
-ln -snf ${UMBRELLA_DATA}${ensindexstr}/mpassit/mpassit.${timestr}.nc .
+cpreq=${cpreq:-cpreq}
+cd ${DATA}
+#
+#  cpy excutable and fix files; decide mesh
+#
+${cpreq} ${EXECrrfs}/upp.x .
 ${cpreq} ${FIXrrfs}/upp/* .
 FIXcrtm=${FIXrrfs}/crtm/2.4.0
 while read line; do
   ln -snf ${FIXcrtm}/${line} .
 done < crtmfiles.upp
+#
+#  find cycle time
+#
+YYYYMMDDHH=${CDATE}
```

Comment:
1. Why do we re-define `CDATE` to `YYYYMMDDHH`?
2. `YYYYMMDDHH` is not used anywhere

Reply:
This is the format used by RRFSv1. It is more clear what the time looks like. I just added those and think about to use them in later update.

Reply:
I don't think it is a good idea to define a new variable that is already defined by the NCO standard.

Reply:
In the previous RAP/HRRR/RRFSv1 era,  we defined lots of YYYYMMDDHH-ish variables. In my view, these are not good practices because the [NCO standard ](https://www.nco.ncep.noaa.gov/idsb/implementation_standards/ImplementationStandards.v11.0.0.pdf) has defined those built-in variables:
```
CDATE=YYYYMMDDHH
PDY=YYYYMMDD
cyc=HH
cycle=tHHz or tHHMMz
subcycle=MM
```
I don't think we need to introduce new variables for those existing things.

If we want to extract some parts from the date string, such as `2024112700`,  temporarily, we can always use the BASH substring functionality easily, for example, ${CDATE:0:6} will be `202411`.  This will help us stick with the NCO standards. Just my two cents.  

With that said, I am okay with the majority vote, no matter if it is `CDATE` or `YYYYMMDDHH`.

@MatthewPyle-NOAA @BenjaminBlake-NOAA What do you think? Thanks!

Reply:
We can discuss this later and make decision how to update all of the scripts if needed.

Reply:
To add to this discussion, `CDATE, PDY, cyc, cycle, subcyc`  have strict definitions and they hold and can only hold the values for the current cycle. 
On the other hand, `YYYYMMDDHH`-ish variables are not strictly tied to the current cycle, people can use them for any previous or future cycles and that will confuse users when reading the scripts.